packages/internal/src/tmpDir.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Create a `tmpDir` function that synchronously creates a temporary directory
+ * and a function for deleting it along with any added contents.
+ *
+ * @param {Pick<import('tmp'), 'dirSync'>} tmp
+ */
+export const makeTempDirFactory = tmp => {
+  /** @type {(prefix?: string) => [dirName: string, cleanup: () => void]} */
+  const tmpDir = prefix => {
+    const { name, removeCallback } = tmp.dirSync({
```

Comment:
While I agree there is likely no need to use the async version here, another effect is that `cleanup` is now also sync. I don't think it matters, just that it might create some unnecessary `await`.